### PR TITLE
Improve NewPipe's sharing and fix crash when opening content in browser on some devices

### LINKED ISSUE #3925: Error while opening content in browser
### Version

0.19.6

### Steps to reproduce the bug

1. Go to 'info about video'
2. Press on 'open in browser'

### Expected behavior

Opening url in browser.

### Actual behaviour

UI crashes

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ ru
* __Service:__ none
* __Version:__ 0.19.6
* __OS:__ Linux Android 7.0 - 24

<details><summary><b>Crash log</b></summary><p>

```
android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=https://www.youtube.com/... pkg=com.huawei.android.internal.app }
	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1854)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1544)
	at android.app.Activity.startActivityForResult(Activity.java:4391)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:675)
	at android.app.Activity.startActivityForResult(Activity.java:4335)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:662)
	at android.app.Activity.startActivity(Activity.java:4697)
	at android.app.Activity.startActivity(Activity.java:4665)
	at org.schabi.newpipe.util.ShareUtils.openUrlInBrowser(ShareUtils.java:36)
	at org.schabi.newpipe.fragments.detail.VideoDetailFragment.onOptionsItemSelected(VideoDetailFragment.java:702)
	at androidx.fragment.app.Fragment.performOptionsItemSelected(Fragment.java:2830)
	at androidx.fragment.app.FragmentManager.dispatchOptionsItemSelected(FragmentManager.java:2717)
	at androidx.fragment.app.FragmentController.dispatchOptionsItemSelected(FragmentController.java:412)
	at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:389)
	at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:219)
	at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
	at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
	at androidx.appcompat.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:64)
	at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:207)
	at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:781)
	at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:840)
	at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:991)
	at androidx.appcompat.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
	at android.widget.AdapterView.performItemClick(AdapterView.java:313)
	at android.widget.AbsListView.performItemClick(AbsListView.java:1203)
	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3197)
	at android.widget.AbsListView$3.run(AbsListView.java:4140)
	at android.os.Handler.handleCallback(Handler.java:761)
	at android.os.Handler.dispatchMessage(Handler.java:98)
	at android.os.Looper.loop(Looper.java:156)
	at android.app.ActivityThread.main(ActivityThread.java:6617)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)

```
</p></details>
<hr>

### PR DESCRIPTION
**Pull request by TiA4f8R**
#### What is it?
- [x] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

This PR fixes the crash when no browser is set as default on some devices (NewPipe thinks that the system app chooser is a browser because its package ID isn't "android" due to some OEM's system customizations) by catching the ActivityNotFound exception and uses the openInDefaultApp method. I also made improvements in NewPipe's sharing by trying to use Android's system share sheet instead of using OEM's share sheet (I used this last as a fallback if this first has been removed) and also made it for opening content in browser. Here is the difference between the original code and my changes (my device is an Honor 9X, running EMUI 10):

| Before this PR | After this PR |
|----------------|---------------|
| Share panel before               | Share panel after              |
| Share panel before this PR               | Share panel after this PR              |
| Open in browser before               | Open in browser after              |
| Open in browser before this PR              | Open in browser after this PR              |

I also updated NewPipe's URLs (API and links in AboutActivity) with the new domain name.

#### Fixes the following issue(s)
Fixes #3925

#### APK testing 

[NewPipe fix-crash-open-in-browser.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/5654658/NewPipe.fix-crash-open-in-browser.apk.zip)

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

**Important note: I am not an Android / Java developer** so please be indulgent with me.

### CODE REVIEW

**Comment by TobiGr** [comment]
thank you for the pull request! It looks like you messed up the git history. please drop 074987f40b7a9662234da5f92ee08d05d03cd433 and rebase your branch on our `dev` branch afterwards.

**Comment by TiA4f8R** [comment]
@TobiGr Thanks for the comment. However, I am new to GitHub, so can you guide me (sorry for asking this)?

**Comment by TiA4f8R** [comment]
@TobiGr Done

**Comment by TiA4f8R** [comment]
Where is the code to open links from descriptions and comments? I can't found it so my improvements aren't applied to those links.

**Comment by TobiGr** [comment]
Ah, seems like you've figured it out yourself! You can use [`git reset`]([slightly different](/uploads/992efb094555450243f34c7e2bb12729/960.html)) to remove 2932eef5875a171007f55353918da86f487157c6.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -685,4 +685,5 @@
     <string name="playlist_page_summary">Playlist page</string>
     <string name="show_thumbnail_title">Show thumbnail</string>
     <string name="show_thumbnail_summary">Show thumbnail on lock screen as background and inside notifications</string>
+    <string name="open_with">Open with</string>
```
Comment (TobiGr) [inline app/src/main/res/values/strings.xml:688]: please move this to the other "open ..." strings (line 12f).

**Review by TobiGr — COMMENTED** [review]
Thank you.
Please open a separate PR for unrelated changes next time. That speeds the reviewing process up.

**Comment by TiA4f8R** [comment]
@TobiGr Changes applied

**Comment by Stypox** [comment]
It seems like there is still something wrong with the commit history of this PR.
@B0pol could you take care of this, afterwards, since you improved sharing functions a while ago?

**Comment by B0pol** [comment]
@Stypox I'll look at this the next weekend

**Comment by TiA4f8R** [comment]
@Stypox @B0pol Can I reopen a proper PR in a new branch when I have free time (not in a near future)?
(I am new to open source projects and very bad with Git for now, sorry :( .)

**Comment by Stypox** [comment]
@TiA4f8R don't worry, we have all been new to git once ;-) 
If you need any help feel free to ask :-D

**Comment by TiA4f8R** [comment]
@Stypox The last commits are from a rebase on the dev branch but I had merging conflicts with one of my modified files so I force-pushed but I didn't think it will do this.

**Pull request closed without merging**